### PR TITLE
Posts Controller Headers and Links Fixes

### PR DESCRIPTION
**Pull request by rachelbaker**

### CODE REVIEW

**Comment by rachelbaker** [comment]
Fixes #738

**Comment by rachelbaker** [comment]
@WP-API/amigos #reviewmerge

File: lib/class-wp-json-posts-controller.php [hunk 1]
```diff
@@ -748,6 +747,22 @@ protected function check_delete_permission( $post ) {
...
 	/**
+	 * Get the base path for a post type's endpoints.
+	 *
+	 * @param object|string $post_type
+	 * @return string       $base
+	 */
+	protected function get_post_type_base( $post_type ) {
+		if ( ! is_object( $post_type ) ) {
+			$post_type = get_post_type_object( $post_type );
+		}
```
Comment (danielbachhuber) [inline lib/class-wp-json-posts-controller.php]: Do we really need this?

File: lib/class-wp-json-posts-controller.php [hunk 2]
```diff
@@ -748,6 +747,22 @@ protected function check_delete_permission( $post ) {
...
+
+		$base = ! empty( $post_type->json_base ) ? $post_type->json_base : $post_type->name;
```
Comment (danielbachhuber) [inline lib/class-wp-json-posts-controller.php]: Crafty! We should probably have `register_post_type()` fill in `$post_type->json_base` so we don't need this conditional.

File: lib/class-wp-json-posts-controller.php (hunk 1 above)
Comment (danielbachhuber) [inline lib/class-wp-json-posts-controller.php]: Eh, I'm fine with it.

**Pull request merged**